### ISSUE TITLE
kubectl Cheat Sheet: zsh autocomplete

### ISSUE THREAD

**Issue by cpboyd** [kubernetes/website#33929]
The zsh section says:
```zsh
source <(kubectl completion zsh)  # setup autocomplete in zsh into the current shell
echo "[[ $commands[kubectl] ]] && source <(kubectl completion zsh)" >> ~/.zshrc # add autocomplete permanently to your zsh shell
```

Should this be `echo "[[ '$commands[kubectl]' ]] && source <(kubectl completion zsh)" >> ~/.zshrc`?

On macOS, if `kubectl` doesn't exist and the script has `[[  ]]` then it throws an error.
If, instead, it has `[[ '' ]]` then it short-circuits the `source` properly as expected, without error.

**Comment by cpboyd** [comment]
Maybe also note `autoload -U compinit && compinit` as compinit isn't included with the macOS zsh by default

**Comment by Sea-n** [comment]
Maybe this is better:

```zsh
echo '[[ "$commands[kubectl]" ]] && source <(kubectl completion zsh)' >> ~/.zshrc
```

Don't interpret `$commands[kubectl]` beforehand, and short-circuit when it's not exists.

**Comment by cpboyd** [comment]
@Sea-n Yeah. That's probably true, since they'd have to run it anytime the path changed as well.
I'm currently using 1.22 as the latest isn't compatible with my server's version (but other homebrew recipes like awscli depend on the latest)

**Comment by sftim** [comment]
What's wrong with the existing text? I'm not clear on what the shortcoming is.

**Comment by sftim** [comment]
OK, so instead of the existing text
> ```shell
> source <(kubectl completion zsh)  # setup autocomplete in zsh into the current shell
> echo "[[ $commands[kubectl] ]] && source <(kubectl completion zsh)" >> ~/.zshrc # add autocomplete permanently to your zsh shell
> ```



we could have:
```shell
source <(kubectl completion zsh)  # setup autocomplete in zsh into the current shell
echo '[[ $commands[kubectl] ]] && source <(kubectl completion zsh)' >> ~/.zshrc # add autocomplete permanently to your zsh shell
```

Does that sound right?

**Comment by cpboyd** [comment]
@sftim Yes, that should work

**Comment by Sea-n** [comment]
Already fixed in PR #34579.
/assign @jjinux
/remove-lifecycle rotten
/close
